packages/server/modules/workspaces/services/join.ts

```diff
@@ -17,15 +16,13 @@ export const joinWorkspaceFactory =
   ({
     getUserEmails,
     getWorkspaceWithDomains,
-    upsertWorkspaceRole,
     emitWorkspaceEvent,
-    ensureValidWorkspaceRoleSeat
+    addOrUpdateWorkspaceRole
   }: {
     getUserEmails: FindEmailsByUserId
     getWorkspaceWithDomains: GetWorkspaceWithDomains
-    upsertWorkspaceRole: UpsertWorkspaceRole
+    addOrUpdateWorkspaceRole: AddOrUpdateWorkspaceRole
```

Comment:
@cdriesler @Mikehrn what's the difference between join() and requestToJoin()? can we test out both mutations?